lib/ansible/modules/system/user.py

```diff
@@ -695,15 +704,16 @@ def modify_user(self):
         return self.modify_user_usermod()
 
     def create_homedir(self, path):
+        self.parse_defaults()
         if not os.path.exists(path):
             if self.skeleton is not None:
                 skeleton = self.skeleton
             else:
-                skeleton = '/etc/skel'
+                self.skeleton = '/etc/skel'
```

Comment:
`skeleton` requires `createhome` but this isn't enforced. `required_if` should be used with `AnsibleModule`.